File: Frontend/StudyVera.FrontEnd/StudyVera.FrontEnd/Services/ProfileStatService.cs
using Blazored.LocalStorage;
using StudyVera.FrontEnd.Models.LessonSchedules;
using StudyVera.FrontEnd.Models.ProfileStats;
using StudyVera.FrontEnd.Services.Concrats;
using StudyVera.FrontEnd.Services.Helpers;
using StudyVera.FrontEnd.Utilities;
using System.Net.Http.Json;

namespace StudyVera.FrontEnd.Services;

public class ProfileStatService : ServiceHelper, IProfileStatService
{
    private readonly HttpClient _httpClient;
    private readonly string _apiUrl = $"{AppConsts.ApiBaseUrl}profile/stats/";
    public ProfileStatService(ILocalStorageService localStorage, HttpClient client) : base(localStorage, client)
    {
        _httpClient = client;
    }

    public async Task<ProfileStatDto> GetScoreAsync()
    {
        await AddAuthorizationHeader();
        try
        {
            var response = await _httpClient.GetAsync(_apiUrl + "me");
        }
        catch (Exception)
        {

            throw;
        }
        throw new NotImplementedException();
    }

    public async Task<List<ScoreBoardDto>> GetScoreBoardAsync()
    {
        await AddAuthorizationHeader();
        try
        {
            var response = await _httpClient.GetAsync($"{_apiUrl}?PageNumber=1&PageSize=5");
            if (response.IsSuccessStatusCode)
            {
                var scoreBoardResult = await response.Content.ReadFromJsonAsync<List<ScoreBoardDto>>();
                if (scoreBoardResult != null)
                    return scoreBoardResult;

            }
        }
        catch (Exception)
        {
            Console.WriteLine("Veri alınırken hata oluştu!. lütfen daha sonra tekrar deneyiniz.");
            return new List<ScoreBoardDto>();
        }
        return new List<ScoreBoardDto>();
    }
}

File: Frontend/StudyVera/src/app/services/profile-stat.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { environment } from '../../environments/environment';
import { ProfileStatDto, ScoreBoardDto } from '../models/profile-stat.model';

@Injectable({
  providedIn: 'root'
})
export class ProfileStatService {
  private http = inject(HttpClient);
  
  private apiUrl = `${environment.apiUrl}/profile/stats`;

  getScore(): Observable<ProfileStatDto | null> {
    return this.http.get<ProfileStatDto>(`${this.apiUrl}/me`).pipe(
      catchError((error) => {
        console.error('ProfileStatService.getScore error:', error);
        return of(null); 
      })
    );
  }

  getScoreBoard(pageNumber: number = 1, pageSize: number = 5): Observable<ScoreBoardDto[]> {
    const params = new HttpParams()
      .set('PageNumber', pageNumber.toString())
      .set('PageSize', pageSize.toString());

    return this.http.get<ScoreBoardDto[]>(this.apiUrl, { params }).pipe(
      catchError((error) => {
        console.error('Veri alınırken hata oluştu! Lütfen daha sonra tekrar deneyiniz.', error);
        return of([]);
      })
    );
  }
}